examples/Receiver_Response.py

```diff
@@ -0,0 +1,164 @@
+    thisdata /= 1e6
+    thisdata.rename("nu_test", r"$\Delta \nu$")
+    thisdata.name(r"$\mathrm{dg_{%s\ \mathrm{kHz}}}/ \mu V$" % SW)
+    return thisdata
+
+
+file1 = "240123_10mV_AFG_GDS_5mV_100MSPS_analytic.h5"
+file2 = "240117_afg_sc_10mV_3p9kHz_zoom.h5"
+lambda_G = 0.4e3  # Width for Gaussian convolution
+# There are less nodes acquired for the control case (since we assume it's
+# relatively flat) so I need to separately define them
+file1_nodes = psd.find_file(
+    re.escape(file1),
+    exp_type="ODNP_NMR_comp/noise_tests",
+    return_list=True,
+)
+control_frqs_kHz = np.array([float(j.split("_")[1]) for j in file1_nodes])
+file2_nodes = psd.find_file(
+    re.escape(file2),
+    exp_type="ODNP_NMR_comp/noise_tests",
+    return_list=True,
+)
+# $\nu_{test}$
+resp_frq_kHz = np.array([float(j.split("_")[1]) for j in file2_nodes])
+# {{{ Calculate input power (acquired on Oscilloscope)
+#     To determine the input power, just take a capture of the signal in the
+#     time domain and fit to a complex exponential
+# {{{ Make empty nddata to drop the calculated $V^{2}$ into with corresponding
+#     frequency ($\nu$) output by AFG source, and set the frequencies based on
+#     the node names
+control = psd.ndshape([len(control_frqs_kHz)], ["nu_test"]).alloc()
+control.setaxis("nu_test", control_frqs_kHz).set_units("nu_test", "Hz")
+# }}}
+for j, nodename in enumerate(file1_nodes):
+    rf_frq = control["nu_test"][j]
+    d = psd.find_file(
+        file1,
+        expno=nodename,
+        exp_type="ODNP_NMR_comp/noise_tests",
+    )
+    # {{{ fit signal in t domain to complex exponential
+    A, omega, phi, t = symbols("A omega phi t", real=True)
+    f = psd.lmfitdata(d)
+    f.functional_form = A * sp.exp(1j * 2 * np.pi * omega * t + 1j * phi)
+    f.set_guess(
+        A=dict(value=5e-2, min=1e-4, max=1),
+        omega=dict(
+            value=rf_frq,
+            min=rf_frq - 1e4,
+            max=rf_frq + 1e4,
+        ),
+        phi=dict(value=0.75, min=-np.pi, max=np.pi),
+    )
+    f.fit(use_jacobian=False)
+    fit = f.eval()
+    # }}}
+    V_amp = f.output("A")
+    control["nu_test", j] = V_amp**2
+# {{{ make spline for power going into RX box
+control.sort("nu_test")
+Pin_spline = control.spline_lambda()
+# }}}
+# }}}
+# {{{ Calculate $dg^2$
+for j, nodename in enumerate(file2_nodes):
+    d = psd.find_file(
+        file2,
+        exp_type="ODNP_NMR_comp/noise_tests",
+        expno=nodename,
+    )
+    carrier = (
+        d.get_prop("acq_params")["carrierFreq_MHz"] * 1e6
+    )  # $\nu_{RX,LO}$
+    if j == 0:
+        # Allocate and array that's shaped like the data acquired
+        # for one frequency but add axis to store the frequency of
+        # the test signal frequency
+        rec_data1 = (d.shape + ("nu_test", len(resp_frq_kHz))).alloc()
+        rec_data1.setaxis("t", d.getaxis("t")).set_units("t", "s")
+        rec_data1.setaxis("nu_test", resp_frq_kHz).set_units("nu_test", "Hz")
```

Comment:
need to allocate shape in here to have the nScans axis (inspired by how we allocate for saving the psd data in the acquisition files